include/zephyr/drivers/adc.h

```diff
@@ -681,6 +681,21 @@ static inline int z_impl_adc_read(const struct device *dev,
 	return api->read(dev, sequence);
 }
 
+/**
+ * @brief Set a read request from a struct adc_dt_spec.
+ *
+ * @param spec ADC specification from Devicetree.
+ * @param sequence  Structure specifying requested sequence of samplings.
+ *
+ * @return A value from adc_read().
+ * @see adc_read()
+ */
+static inline int adc_read_dt(const struct adc_dt_spec *spec,
```

Comment:
This I think should be in another PR because start to mix the goals.

Reply:
Moved here https://github.com/zephyrproject-rtos/zephyr/pull/63010